### ISSUE TITLE
How to disclose the GeoSPARQL capabilities of SPARQL endpoints

### ISSUE THREAD

**Issue by FransKnibbe** [opengeospatial/ogc-geosparql#310]
In an automated web of data, it would be good if consumers, especially machines, have a way of establishing if and how a SPARQL endpoint support GeoSPARQL, in the form of an exchange of data.

For instance, the GeoSPARQL specification could include guidance on how to use [SPARQL Service Description](https://www.w3.org/TR/sparql11-service-description/) to make the following known:

1. Whether the endpoint supports GeoSPARQL
2. Which version(s) of GeoSPARQL are supported
3. Which GeoSPARQL modules (conformance classes) are supported

**Comment by situx** [comment]
That is a very good idea.
We should definitely follow up on this.
Currently, especially in my software implementations, I use a set of test queries to find out whether GeoSPARQL is supported and to which extent, but a service description would help a lot.

**Comment by situx** [comment]
I had a look at the specification vocabulary today and I will break down my insights here:

Insight number 1:
You are able to specify a query language type, which would be easy for us to do and is just one triple.
sd:supportedLanguage sd:SPARQL11Query ; is an example for SPARQL 1.1.
We could simply define two URIs for GeoSPARQL1.0 and GeoSPARQL 1.1 query language, right?
However,  I believe we currently do not have either to identify the query language itself.
We have:
* <http://www.opengis.net/def/geosparql/1.1> to identify the standard as a whole
* <http://www.opengis.net/def/geosparql/funcsrules> to identify functions and rules
So I wonder if at least a SPARQL service description with a unique URL identifying the GeoSPARQL query language is feasible.
@nicholascar do we really not have a URI to identify the query language itself as of now?


Insight number 2:
The service description allows only the specification of extension functions, not modules.
To me, it seems extension functions are only defined if they are not included in the specified supported query languages.
However, we have URIs for every query function we define in GeoSPARQL in the requirements vocabulary, so it would be no stretch to add them to a service description.

Modules are not defined in the service description vocabulary, but we already have URIs defined for modules and could assign them to the sd:feature property, as is done for example to indicate the ability for federated querying.

To sum up, we would need:
* The definition of two URIs for the GeoSPARQL 1.0 and GeoSPARQL 1.1 query languages
* A decision on whether we recommend listing of all GeoSPARQL functions in a service description
* A decision on whether it makes sense to list GeoSPARQL modules as sd:feature

If these decisions are taken, I think creating a service description even for GeoSPARQL 1.1 is really not too much work.
Any opinions? @nicholascar @mperry455 @FransKnibbe @dr-shorthair
